flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -648,6 +700,29 @@ private static void start(YarnConfiguration conf, String principal, String keyta
 
 			File targetTestClassesFolder = new File("target/test-classes");
 			writeYarnSiteConfigXML(conf, targetTestClassesFolder);
+
+			LOG.info("Starting up MiniDFSCluster");
+			if (miniDFSCluster == null) {
+
+				Configuration hdfsConfiguration = new Configuration();
+				hdfsConfiguration.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, tmpHDFS.getRoot().getAbsolutePath());
+				populateHDFSSecureConfigurations(hdfsConfiguration, principal, keytab);
+				MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(hdfsConfiguration);
+
+				int port = NAME_NODE_START_PORT;
```

Comment:
I do not think we need to check the available port. MiniDFSCluster will use a random port to start namenode, and we just use `hdfsCluster.getURI()` to get the schema. Then set `fs.defaultFS` in hadoop configuration.

Reply:
Are you right. It is not needed. Thanks for the suggestion.